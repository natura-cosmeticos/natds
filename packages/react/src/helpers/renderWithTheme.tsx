import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
