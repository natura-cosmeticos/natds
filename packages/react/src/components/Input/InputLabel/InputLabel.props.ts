import { InputProps } from '../Input.props'

export type InputLabelProps = Pick<InputProps, 'id' | 'label' | 'required' | 'feedback' | 'disabled'>
