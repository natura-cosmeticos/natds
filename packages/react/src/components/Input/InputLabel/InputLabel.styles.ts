import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { InputLabelProps } from './InputLabel.props'

export const getFeedbackTextColor = (theme: Theme) => ({ feedback, disabled }: InputLabelProps) => {
  const defaultColor = disabled ? theme.color.lowEmphasis : theme.color.mediumEmphasis

  switch (feedback) {
    case 'error':
      return theme.color.alert
    case 'success':
      return theme.color.success
    default:
      return defaultColor
  }
}

const styles = createUseStyles((theme: Theme) => ({
  label: {
    color: getFeedbackTextColor(theme),
    display: ({ label }: InputLabelProps) => (label ? 'inline-block' : 'none'),
    fontFamily: [theme.typography.fontFamily.primary, theme.typography.fontFamily.secondary],
    fontSize: 14,
    letterSpacing: 0.11,
    marginBottom: theme.spacing.micro
  }
}))

export default styles
