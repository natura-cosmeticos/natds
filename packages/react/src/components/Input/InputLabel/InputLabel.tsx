import React from 'react'
import { InputLabelProps } from './InputLabel.props'
import styles from './InputLabel.styles'

const InputLabel = ({
  id,
  label,
  required,
  feedback,
  disabled
}: InputLabelProps): JSX.Element => {
  const classes = styles({ feedback, disabled, label })

  return (
    <label htmlFor={id} className={classes.label}>
      {label}
      {required && label && '*'}
    </label>
  )
}

export default InputLabel
