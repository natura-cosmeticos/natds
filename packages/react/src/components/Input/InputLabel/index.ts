import InputLabel from './InputLabel'

export { InputLabel }
export default InputLabel
