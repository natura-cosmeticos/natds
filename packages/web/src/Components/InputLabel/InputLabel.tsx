import * as React from 'react'
import MaterialInputLabel from '@material-ui/core/InputLabel'
import { IInputLabelProps } from './InputLabel.props'
import useStyles from './InputLabel.styles'

export { IInputLabelProps } from './InputLabel.props'

export const InputLabel = React.forwardRef<HTMLLabelElement, IInputLabelProps>(
  (props: IInputLabelProps, ref) => {
    const { state, ...otherProps } = props
    const classes = useStyles({ state })

    return (
      <MaterialInputLabel
        classes={classes}
        shrink
        ref={ref}
        {...otherProps}
      />
    )
  }
)

InputLabel.displayName = 'InputLabel'

export default InputLabel
