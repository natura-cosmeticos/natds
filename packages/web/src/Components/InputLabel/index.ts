import InputLabel from './InputLabel'

export { IInputLabelProps } from './InputLabel.props'
export { InputLabel }
export default InputLabel
