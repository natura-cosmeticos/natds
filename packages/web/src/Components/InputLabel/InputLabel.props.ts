import { InputLabelProps } from '@material-ui/core/InputLabel'
import { State } from '../Input/Input.props'

export interface IInputLabelProps extends Omit<InputLabelProps, 'shrink'> {

  /**
   * The contents of the `InputLabel`.
   *
   * @optional
   * @type node
   */
  children?: InputLabelProps['children']

  /**
   * Override or extend the styles applied to the component.
   *
   * See [CSS API](https://material-ui.com/pt/api/input-label/#css) for more details.
   *
   * @optional
   * @see https://material-ui.com/pt/api/input-label/#css
   * @type object
   */
  classes?: InputLabelProps['classes']

  /**
   * The color of the component. It supports those theme colors that make sense for this component.
   *
   * @optional
   * @type "primary" | "secondary"
   */
  color?: InputLabelProps['color']

  /**
   * If `true`, the transition animation is disabled.
   *
   * @default false
   * @optional
   * @type bool
   */
  disableAnimation?: InputLabelProps['disableAnimation']

  /**
   * If `true`, apply disabled class.
   *
   * @optional
   * @type bool
   */
  disabled?: InputLabelProps['disabled']

  /**
   * If `true`, the label will be displayed in an error state.
   *
   * @optional
   * @type bool
   */
  error?: InputLabelProps['error']

  /**
   * If `true`, the input of this label is focused.
   *
   * @optional
   * @type bool
   */
  focused?: InputLabelProps['focused']

  /**
   * If `dense`, will adjust vertical spacing. This is normally obtained via context from `FormControl`.
   *
   * @optional
   * @type "dense"
   */
  margin?: InputLabelProps['margin']

  /**
   * if `true`, the label will indicate that the input is required.
   *
   * @optional
   * @type bool
   */
  required?: InputLabelProps['required']

  /**
   * @optional
   * @type "error" | "success" | undefined
   */
  state?: State

  /**
   * The variant to use.
   *
   * @optional
   * @type "filled" | "outlined" | "standard"
   */
  variant?: InputLabelProps['variant']
}
