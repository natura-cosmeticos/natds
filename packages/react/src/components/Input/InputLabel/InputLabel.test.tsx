import React from 'react'
import renderWithTheme from '../../../helpers/renderWithTheme'
import InputLabel from '.'
import { InputLabelProps } from './InputLabel'

const defaultProps: InputLabelProps = {
  id: 'label',
  label: 'some label'
}

describe('Input component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<InputLabel {...defaultProps} />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when is required', () => {
    const { styles, component } = renderWithTheme(<InputLabel {...defaultProps} required />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when there is an error feedback', () => {
    const { styles, component } = renderWithTheme(<InputLabel {...defaultProps} feedback="error" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly when there is an success feedback', () => {
    const { styles, component } = renderWithTheme(<InputLabel {...defaultProps} feedback="success" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
