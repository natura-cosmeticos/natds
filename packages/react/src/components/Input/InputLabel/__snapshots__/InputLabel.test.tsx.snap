// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly when is required 1`] = `
Array [
  ".label-0-2-1 {
  font-size: 14px;
  font-family: Roboto, sans-serif;
  margin-bottom: 4px;
  letter-spacing: 0.11px;
}
.label-d0-0-2-2 {
  color: #777777;
  display: inline-block;
}",
  <div>
    <label
      class="label-0-2-1 label-d0-0-2-2"
      for="label"
    >
      some label
      *
    </label>
  </div>,
]
`;

exports[`Input component should render correctly when there is an error feedback 1`] = `
Array [
  ".label-0-2-1 {
  font-size: 14px;
  font-family: Roboto, sans-serif;
  margin-bottom: 4px;
  letter-spacing: 0.11px;
}
.label-d0-0-2-2 {
  color: #e74627;
  display: inline-block;
}",
  <div>
    <label
      class="label-0-2-1 label-d0-0-2-2"
      for="label"
    >
      some label
    </label>
  </div>,
]
`;

exports[`Input component should render correctly when there is an success feedback 1`] = `
Array [
  ".label-0-2-1 {
  font-size: 14px;
  font-family: Roboto, sans-serif;
  margin-bottom: 4px;
  letter-spacing: 0.11px;
}
.label-d0-0-2-2 {
  color: #569a32;
  display: inline-block;
}",
  <div>
    <label
      class="label-0-2-1 label-d0-0-2-2"
      for="label"
    >
      some label
    </label>
  </div>,
]
`;

exports[`Input component should render correctly with default props 1`] = `
Array [
  ".label-0-2-1 {
  font-size: 14px;
  font-family: Roboto, sans-serif;
  margin-bottom: 4px;
  letter-spacing: 0.11px;
}
.label-d0-0-2-2 {
  color: #777777;
  display: inline-block;
}",
  <div>
    <label
      class="label-0-2-1 label-d0-0-2-2"
      for="label"
    >
      some label
    </label>
  </div>,
]
`;
